src/libraries/System.Private.CoreLib/src/System/IO/FileSystem.Unix.cs

```diff
@@ -302,7 +302,8 @@ public static void CreateDirectory(string fullPath)
             string errorString = fullPath;
             while (stackDir.Count > 0)
```

Comment:
```
for (int i = stackDir.Count - 1; i >= 0; i--)
{
    string name = stackDir[i];
...
```
I do not think you need to spend time on actually removing the items from the list.

Reply:
Or you can get rid of the list alltogether, and just go through the path and look for directory seperators.